cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -316,20 +316,23 @@ public KafkaAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
                 .compose(state -> state.kafkaExporterService())
                 .compose(state -> state.kafkaExporterReady())
 
-                .compose(state -> {
-                    chainPromise.complete();
-                    return chainPromise.future();
+                .compose(state -> Future.succeededFuture())
```

Comment:
What is this compose doing?

Reply:
I do not remember if this is just a left over line or if it has reason. I would need to check when I get back home.